src/inventory/edit/EditItem.tsx

```diff
@@ -0,0 +1,92 @@
+import { Spinner, Button, Toast } from '@hospitalrun/components'
+import _ from 'lodash'
+import React, { useEffect, useState } from 'react'
+import { useHistory, useParams } from 'react-router-dom'
+
+import useAddBreadcrumbs from '../../page-header/breadcrumbs/useAddBreadcrumbs'
+import { useUpdateTitle } from '../../page-header/title/TitleContext'
+import useTranslator from '../../shared/hooks/useTranslator'
+import InventoryItem from '../../shared/model/InventoryItem'
+import useItem from '../hooks/useItem'
+import useUpdateItem from '../hooks/useUpdateItem'
+import ViewItemDetails from '../view/ViewItemDetails'
+
+const EditItem = () => {
+  const { t } = useTranslator()
+  const { id } = useParams()
+
+  const updateTitle = useUpdateTitle()
+  updateTitle(t('inventory.items.edit'))
+  const history = useHistory()
+
+  const [newItem, setItem] = useState({} as InventoryItem)
+  const { data: currentItem, isLoading: isLoadingItem } = useItem(id)
+
+  const {
+    mutate: updateMutate,
+    isLoading: isLoadingUpdate,
+    isError: isErrorUpdate,
+    error: updateMutateError,
+  } = useUpdateItem(newItem)
+
+  useAddBreadcrumbs([
+    {
+      i18nKey: 'inventory.items.edit',
+      location: `/inventory/edit/${id}`,
+    },
+  ])
+
+  useEffect(() => {
+    if (currentItem !== undefined) {
+      setItem(currentItem)
+    }
+  }, [currentItem])
```

Comment:
This is a clever way to make the component state automatically reflect update react query state. I like it.